### PR TITLE
[Components with more than 1 service] LPS-183224 Refactor PortletProvider implementations from Commerce

### PR DESCRIPTION
**Pull request by rafaprax**
Ticket: https://issues.liferay.com/browse/LPS-183224

### CODE REVIEW

**Comment by rafaprax** [comment]
ci:test:commerce

**Comment by rafaprax** [comment]
ci:test:echo

**Comment by rafaprax** [comment]
ci:test:data-engine

**Comment by rafaprax** [comment]
ci:test:relevant

File: portal-kernel/src/com/liferay/portal/kernel/portlet/PortletProviderUtil.java [hunk 1]
```diff
@@ -148,29 +132,80 @@ protected static boolean isAssetObject(String className) {
...
+		<PortletProvider.Action, Map<String, PortletProvider>>
+			_portletProviderActionPortletProviderMap =
+				new ConcurrentHashMap<>();
+	private static final ServiceTracker<PortletProvider, PortletProvider>
+		_portletProviderServiceTracker;
+
+	static {
+		BundleContext bundleContext = SystemBundleUtil.getBundleContext();
+
+		_portletProviderServiceTracker = new ServiceTracker<>(
```
Comment (tinatian) [inline portal-kernel/src/com/liferay/portal/kernel/portlet/PortletProviderUtil.java]: @rafaprax For this service tracker, let's do this. Only track PortletProvider, when you want to retrieve portlet provider for a specific action: 1 get the portlet provider 2 if provider exists, check its supported actions include the one you asked.

**Comment by Kvale1733** [comment]
ci:test:commerce

**Comment by Kvale1733** [comment]
ci:test:data-engine

**Comment by Kvale1733** [comment]
ci:test:echo

**Comment by Kvale1733** [comment]
ci:test:relevant

**Comment by rafaprax** [comment]
Closing since it will be send in two different prs

**Pull request closed without merging**